editor/blocks/list/index.js

```diff
@@ -0,0 +1,43 @@
+const Editable = wp.blocks.Editable;
+const { html, prop } = wp.blocks.query;
+
+function List( { nodeName, children } ) {
+	// nodeName.toLowerCase() is used to map DOM nodeName values to proper tag.
+	return wp.element.createElement( nodeName.toLowerCase(), null, children );
+}
+
+wp.blocks.registerBlock( 'core/list', {
+	title: 'List',
+	icon: 'editor-ul',
+	category: 'common',
+
+	attributes: {
+		listType: prop( 'ol,ul', 'nodeName' ),
+		items: wp.blocks.query.query(
+			'li',
+			{
+				value: html()
+			}
+		)
+	},
+
+	edit( { attributes, onChange } ) {
+		const { listType = 'ol', items = [] } = attributes;
```

Comment:
When not default, `listType` will be uppercased (`nodeName`). Should the default be uppercased as well? See note below about `Editable` `tagName`.